test/controllers/endorseCompanyClosureCertificate.controller.test.ts

```diff
@@ -56,4 +68,41 @@ describe('EndorseCompanyClosureCertificateController', () => {
       assert.isTrue(htmlAssertHelper.hasText('span#applicantName', 'John Smith'))
     })
   })
+
+  describe('POST - ensure form submission is handled correctly', () => {
+    it('should redirect successfully if validator returns no errors', async () => {
+      const testObject = generateEndorseCertificateFormModel()
+
+      when(mockedDissolutionService.approveDissolution(anything(), anything(), anything())).thenResolve()
+      when(mockedFormValidator.validate(deepEqual(testObject), formSchema)).thenReturn(null)
+
+      const app = createApp(container => {
+        container.rebind(FormValidator).toConstantValue(instance(mockedFormValidator))
+        container.rebind(SessionService).toConstantValue(instance(session))
+        container.rebind(DissolutionService).toConstantValue(instance(mockedDissolutionService))
+      })
+
+      await request(app).post(ENDORSE_COMPANY_CLOSURE_CERTIFICATE_URI)
+        .send(testObject)
+        .expect(MOVED_TEMPORARILY)
+        .expect('Location', REDIRECT_GATE_URI)
+    })
+  })
+
+  it('should render view with errors displayed if validator returns errors', async () => {
```

Comment:
This should be included in the POST 'describe' block, as it is outside right now